src/commands/api/apiCache.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ParsedMongoConnectionString } from '../../mongo/mongoConnectionStrings';
+import { DatabaseTreeItem } from '../../vscode-cosmosdb.api';
+
+/**
+ * This cache is used to speed up api calls from other extensions to the Cosmos DB extension
+ * For now, it only helps on a per-session basis
+ */
+const sessionCache: Map<string, DatabaseTreeItem> = new Map();
+
+export function cacheTreeItem(parsedCS: ParsedMongoConnectionString, treeItem: DatabaseTreeItem): void {
+    sessionCache.set(parsedCS.fullId, treeItem);
+}
+
+export function tryGetTreeItemFromCache(parsedCS: ParsedMongoConnectionString): DatabaseTreeItem | undefined {
+    if (sessionCache.has(parsedCS.fullId)) {
+        return sessionCache.get(parsedCS.fullId);
+    } else {
+        return undefined;
+    }
+}
+
+export function removeTreeItemFromCache(expected: ParsedMongoConnectionString): void {
+    if (!expected.databaseName) {
+        // If parsedCS represents an account, iterate and remove all databases that match that account
```

Comment:
Seems like it might be possible to end up with both an account representation and also database representations, i.e. do you also need to check if there's an account cache to delete, too?

Reply:
I think the current code would still work. I'm iterating through the whole cache regardless of if it's a database or account. Anyways I made a few minor changes to hopefully make this more clear.

Admittedly I haven't done a full round of testing for account-level connection strings. Andrii didn't get that "feature" into master yet and I plan to add support in another PR. See error thrown here: https://github.com/Microsoft/vscode-cosmosdb/pull/942/files#diff-bdabf54afc33845cbe50e0fb1f636b7eR25